app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -78,6 +78,38 @@ def reassign_hostgroup
         end
       end
 
+      api :PUT, "/hosts/bulk/assign_taxonomy", N_("Assign organization/location")
+      param_group :bulk_host_ids
+      param :target_organization_id, :number, :desc => N_("ID of the organization to assign the hosts to")
+      param :target_location_id, :number, :desc => N_("ID of the location to assign the hosts to")
+      param :optimistic_import_organization, :bool, N_("Fix organization on mismatch")
+      param :optimistic_import_location, :bool, N_("Fix location on mismatch")
+      def assign_taxonomy
+        if [params[:optimistic_import_organization], params[:optimistic_import_location]].any? { |val| !val }
```

Comment:
Isn't this easier?
```suggestion
        if !params[:optimistic_import_organization] || !params[:optimistic_import_location]
```

But effectively that means both options are mandatory. Why present them as options at all then?

Another thing to consider: is it mandatory to assign both a location and organization? From reading the code (`Taxonomy.where`) I think it deals with `nil` because then `targets` will not have the entry. That implies you don't need `optimistic_import` to be true for that type.

And to be clear, I can see benefits in presenting the option so perhaps it's better to just drop this validation.

Now that also makes me think you can return early if neither `target_organization_id` nor `target_location_id` was given.